src/mlpack/bindings/go/mlpack/capi/arma_util.cpp

```diff
@@ -0,0 +1,251 @@
+/**
+ * @file arma_util.cpp
+ * @author Yasmine Dumouchel
+ *
+ * Utility function for Go to pass gonum object to an Armadillo Object and
+ * vice versa.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "arma_util.h"
+#include "arma_util.hpp"
+#include "cli_util.hpp"
+#include <mlpack/core/util/cli.hpp>
+
+namespace mlpack {
+namespace util {
+
+extern "C" {
+
+/**
+ * Pass Gonum Dense pointer and wrap an Armadillo mat around it.
+ */
+void MLPACK_ToArma_mat(const char *identifier, const double mat[], int row, int col)
+{
+  // Advanced constructor.
+  arma::mat m(const_cast<double*>(mat), row, col, false, true);
+
+  // Set input parameter with corresponding matrix in CLI.
+  SetParam(identifier, m);
+}
+
+/**
+ * Pass Gonum VecDense pointer and wrap an Armadillo rowvec around it.
+ */
+void MLPACK_ToArma_row(const char *identifier, const double rowvec[], int elem)
+{
+  // Advanced constructor.
+  arma::rowvec m(const_cast<double*>(rowvec), elem, false, true);
+
+  // Set input parameter with corresponding row in CLI.
+  SetParam(identifier, m);
+}
+
+/**
+ * Pass Gonum VecDense pointer and wrap an Armadillo colvec around it.
+ */
+void MLPACK_ToArma_col(const char *identifier, const double colvec[], int elem)
+{
+  // Advanced constructor.
+  arma::colvec m(const_cast<double*>(colvec), elem, false, true);
+
+  // Set input parameter with corresponding column in CLI.
+  SetParam(identifier, m);
+}
+
+/**
+ * Return the memory pointer of an Armadillo mat object.
+ */
+void *MLPACK_ArmaPtr_mat(const char *identifier)
+{
+  arma::mat output = CLI::GetParam<arma::mat>(identifier);
+  if (output.is_empty())
+  {
+    return NULL;
+  }
+  void *ptr = GetMemory(output);
+  return ptr;
+}
+
+/**
+ * Return the memory pointer of an Armadillo umat object.
+ */
+void *MLPACK_ArmaPtr_umat(const char *identifier)
+{
+  arma::Mat<double> output = arma::conv_to<arma::Mat<double>>::from(CLI::GetParam<arma::Mat<size_t>>(identifier));
```

Comment:
Hi @rcurtin right now on I am working on Go bindings. Here I am allocating space to `output` directly without passing it from `GetMemory()` is it a correct way or i am some blunder here. If you could help me with this comment. It would be great help to me. Thank You